src/pages/hcp/api-docs/packer/[[...page]].tsx

```diff
@@ -3,159 +3,101 @@
+	getApiDocsStaticProps,
+	getApiDocsStaticPaths,
+	ApiDocsParams,
+} from 'views/api-docs-view/server'
+import { buildApiDocsBreadcrumbs } from 'views/api-docs-view/server/get-api-docs-static-props/utils'
+import { fetchCloudApiVersionData } from 'views/api-docs-view/utils'
+// Components
 import {
-	generateProductLandingSidebarNavData,
-	generateTopLevelSidebarNavData,
-} from 'components/sidebar/helpers'
+	PathTruncationAside,
+	truncatePackerOperationPath,
+} from 'views/api-docs-view/components'
+// Types
+import type { ApiDocsViewProps } from 'views/api-docs-view/types'
+import type { GetStaticPaths, GetStaticProps } from 'next'
 
-import { CustomPageComponent } from 'types/_app'
-import { isDeployPreview } from 'lib/env-checks'
+/**
+ * The product slug is used to fetch product data for the layout.
+ */
+const PRODUCT_SLUG = 'hcp'
+
+/**
+ * The baseUrl is used to generate
+ * breadcrumb links, sidebar nav levels, and version switcher links.
+ */
+const BASE_URL = '/hcp/api-docs/packer'
 
-const productSlug = 'hcp'
-const targetFile = {
+/**
+ * We source version data from a directory in the `hcp-specs-internal` repo.
+ * See `fetchCloudApiVersionData` for details.
+ */
+const GITHUB_SOURCE_DIRECTORY = {
 	owner: 'hashicorp',
 	repo: 'hcp-specs-internal',
-	path: 'specs/cloud-packer-service/stable/2021-04-30/hcp.swagger.json',
+	path: 'specs/cloud-packer-service',
+	ref: 'main',
 }
 
-type ApiDocsPageProps = InferGetStaticPropsType<typeof getStaticProps>
-const ApiDocsPage: CustomPageComponent<ApiDocsPageProps> = ({
-	apiPageProps,
-}) => {
+/**
+ * Render `<ApiDocsView />` with custom operation path truncation.
+ */
+function HcpPackerApiDocsView(props: ApiDocsViewProps) {
 	return (
-		<OpenApiPageContents
-			info={apiPageProps.info}
-			operationCategory={apiPageProps.operationCategory}
-			// Truncate operation paths,
-			// as they are otherwise very difficult to read
-			massageOperationPathFn={(path) =>
-				path.replace(
-					'/packer/2021-04-30/organizations/{location.organization_id}/projects/{location.project_id}',
-					''
-				)
-			}
-			// Add an introductory warning text to each operation
-			// to make a note that the paths have been truncated
-			renderOperationIntro={function PathAside({ data }) {
-				return (
-					<>
-						{/*
-              @TODO replace DevDotContent & <div> with a base UI component
-              https://app.asana.com/0/1202097197789424/1203820006759167/f
-            */}
-						<DevDotContent>
-							<div className="alert alert-info">
-								<strong>Note:</strong> Operation paths have been truncated for
-								clarity. The full path to this operation is:
-							</div>
-						</DevDotContent>
-						<CodeBlock
-							code={data.__path}
-							theme="dark"
-							options={{ showClipboard: true }}
-						/>
-					</>
-				)
-			}}
+		<ApiDocsView
+			{...props}
+			massagePathFn={truncatePackerOperationPath}
+			renderOperationIntro={({ data }: { data: $TSFixMe }) => (
```

Comment:
```suggestion
			renderOperationIntro={({ data }: { data: OperationObjectType }) => (
```
I'm not sure if that's the right type? But I saw it in the function interface in another file.

Reply:
This is perfect, ty ty 💯